src/desktop/apps/auction_reaction/routes.tsx

```diff
@@ -4,60 +4,66 @@ import { routes } from "reaction/Apps/Auction/routes"
 import { stitch } from "@artsy/stitch"
 
 export const bidderRegistration = async (req, res, next) => {
-  if (!res.locals.sd.CURRENT_USER) {
+  /* If this request is from a registration modal (trying to create a bidder),
+     defer to the auction_support app router
+  */
+  if (req.query["accepted-conditions"] === "true") {
+    next()
+  } else if (!res.locals.sd.CURRENT_USER) {
     return res.redirect(
       `/login?redirectTo=${encodeURIComponent(req.originalUrl)}`
     )
-  }
-  try {
-    const {
-      bodyHTML,
-      redirect,
-      status,
-      headTags,
-      scripts,
-      styleTags,
-    } = await buildServerApp({
-      routes,
-      url: req.url,
-      userAgent: req.header("User-Agent"),
-      context: buildServerAppContext(req, res),
-    })
+  } else {
+    try {
+      const {
+        bodyHTML,
+        redirect,
+        status,
+        headTags,
+        scripts,
+        styleTags,
+      } = await buildServerApp({
+        routes,
+        url: req.url,
+        userAgent: req.header("User-Agent"),
+        context: buildServerAppContext(req, res),
+      })
 
-    if (redirect) {
-      res.redirect(302, redirect.url)
-      return
-    }
+      if (redirect) {
+        res.redirect(302, redirect.url)
+        return
+      }
 
-    // Render layout
-    const layout = await stitch({
-      basePath: __dirname,
-      layout:
-        "../../components/main_layout/templates/react_minimal_header.jade",
-      blocks: {
-        head: () => headTags,
-        body: bodyHTML,
-      },
-      locals: {
-        ...res.locals,
-        assetPackage: "auction_reaction",
-        options: {
-          stripev3: true,
+      // Render layout
+      const layout = await stitch({
+        basePath: __dirname,
+        layout:
+          "../../components/main_layout/templates/react_minimal_header.jade",
+        blocks: {
+          head: () => headTags,
+          body: bodyHTML,
         },
-        scripts,
-        styleTags,
-      },
-    })
+        locals: {
+          ...res.locals,
+          assetPackage: "auction_reaction",
+          options: {
+            stripev3: true,
+          },
+          scripts,
+          styleTags,
+        },
+      })
 
-    res.status(status).send(layout)
-  } catch (error) {
-    console.log("(apps/auction_reaction) Error: ", error)
-    if (error.message.includes("Received status code 404")) {
-      const notFoundError: any = new Error("Sale Not Found")
-      notFoundError.status = 404
-      next(notFoundError)
-    } else {
-      next(error)
+      res.status(status).send(layout)
+    } catch (error) {
+      console.log("(apps/auction_reaction) Error: ", error)
+      if (error.message.includes("Received status code 404")) {
+        const notFoundError: any = new Error("Sale Not Found")
+        notFoundError.status = 404
+        next(notFoundError)
```

Comment:
How does `#next` behave when you pass it an exception?

Reply:
It calls the final route handler, which is the [error handler](https://expressjs.com/en/guide/error-handling.html) - see about one page down for how error handlers are defined.

Reply:
Yup, this is indeed correct